cellprofiler_core/utilities/image.py

```diff
@@ -301,16 +304,37 @@ def download_to_temp_file(url):
             Params={'Bucket': bucket_name, 'Key': key.replace("+", " ")}
         )
 
-    from urllib.request import urlopen
-    logging.info(f"Downloading image from {url}")
-    src = urlopen(url)
-    dest_file = tempfile.NamedTemporaryFile(suffix=ext, delete=False, dir=CP_TEMP_DIR.name)
-    try:
-        shutil.copyfileobj(src, dest_file)
-    except Exception as e:
-        logging.error(f"Unable to download image to temp file. {e}")
-        return None
-    finally:
-        dest_file.close()
-    return dest_file.name
+    elif scheme == 'gs':
+            # Get default Google Cloud Storage project ID from environment.
+            if 'GOOGLE_APPLICATION_CREDENTIALS' in os.environ:
+                with open(os.environ['GOOGLE_APPLICATION_CREDENTIALS'], 'r') as fp:
```

Comment:
unnecessary line of code?

Reply:
Why isn't this necessary?

Reply:
This code is opening the file containing the `GOOGLE_APPLICATION_CREDENTIALS` but it isn't making use of it contents. (`fp` is unused)

Reply:
Okay, thank you for explaining. I have remove those lines!